libs/parallelism/executors/include/hpx/executors/dataflow.hpp

```diff
@@ -204,16 +229,17 @@ namespace hpx { namespace lcos { namespace detail {
 
         /// Passes the futures into the evaluation function and
         /// sets the result future.
-        HPX_FORCEINLINE
-        void execute(std::true_type, Futures&& futures)
+        template <typename Futures_>
+        HPX_FORCEINLINE void execute(std::true_type, Futures_&& futures)
         {
             std::exception_ptr p;
 
             try
             {
                 Func func = std::move(func_);
 
-                util::invoke_fused(std::move(func), std::move(futures));
+                util::invoke_fused(
+                    std::move(func), std::forward<Futures_>(futures));
```

Comment:
@aurianer Frankly, I don't remember why we did it. The only explanation I have would be that we wanted to make sure the `func_` dataflow member was guaranteed to be invalid during the function execution. However I can't think of a reason anymore why this would be important.

Reply:
I've changed this to only do one move now.